src/Features/CSharp/Portable/Braces/CSharpBracesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,404 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.AddBraces
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpAddBracesDiagnosticAnalyzer : DiagnosticAnalyzer, IBuiltInAnalyzer
+    {
+        private static readonly LocalizableString s_addBracesLocalizableTitle =
+            new LocalizableResourceString(nameof(FeaturesResources.Add_braces), FeaturesResources.ResourceManager,
+                typeof(FeaturesResources));
+
+        private static readonly LocalizableString s_addBracesLocalizableMessage =
+            new LocalizableResourceString(nameof(WorkspacesResources.Add_braces_to_0_statement), WorkspacesResources.ResourceManager,
+                typeof(WorkspacesResources));
+
+        private static readonly LocalizableString s_removeBracesLocalizableTitle =
+            new LocalizableResourceString(nameof(FeaturesResources.Remove_braces), FeaturesResources.ResourceManager,
+                typeof(FeaturesResources));
+
+        private static readonly LocalizableString s_removeBracesLocalizableMessage =
+            new LocalizableResourceString(nameof(WorkspacesResources.Remove_braces_from_0_statement), WorkspacesResources.ResourceManager,
+                typeof(WorkspacesResources));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(_noneAddBracesDiagnosticDescriptor,
+                                  _infoAddBracesDiagnosticDescriptor,
+                                  _warningAddBracesDiagnosticDescriptor,
+                                  _errorAddBracesDiagnosticDescriptor,
+                                  _noneRemoveBracesDiagnosticDescriptor,
+                                  _infoRemoveBracesDiagnosticDescriptor,
+                                  _warningRemoveBracesDiagnosticDescriptor,
+                                  _errorRemoveBracesDiagnosticDescriptor);
+
+        private readonly DiagnosticDescriptor _noneAddBracesDiagnosticDescriptor;
```

Comment:
@Pilchie On second thought, I'm not sure.  You see, Error, Warning, and Info all creation the diagnostic, lists it in the Error List in VS, and underlines the specified nodes with squiggles.  While Hidden doesn't do the last two, it does create the diagnostic and make it available when the user hovers over the node or presses CTRL-. with the node selected.